session_state.go

```diff
@@ -75,20 +76,39 @@ func (sm *stateMachine) Disconnected(session *session) {
 	}
 }
 
+var msgTypeExecutionReport = []byte("\x0135=8\x01")
+
+func IsExecutionReport(m []byte) bool {
+	return bytes.Contains(m, msgTypeExecutionReport)
+}
+
 func (sm *stateMachine) Incoming(session *session, m fixIn) {
 	sm.CheckSessionTime(session, time.Now())
 	if !sm.IsConnected() {
 		return
 	}
+	var (
+		rawBytesBuffer = m.bytes
+		rawBytes       []byte
+	)
+	if rawBytesBuffer != nil {
+		rawBytes = rawBytesBuffer.Bytes()
+	}
 
-	session.log.OnIncoming(m.bytes.Bytes())
+	session.log.OnIncoming(rawBytes)
 
 	msg := NewMessage()
-	if err := ParseMessageWithDataDictionary(msg, m.bytes, session.transportDataDictionary, session.appDataDictionary); err != nil {
-		session.log.OnEventf("Msg Parse Error: %v, %q", err.Error(), m.bytes)
-	} else {
+	if session.CleanIncomingHotPath && IsExecutionReport(rawBytes) {
```

Comment:
Could you add benchmark and see how much improvement we receive over ParseMessageWithDataDictionary function?

Reply:
Added a benchmark for the whole `session.Incoming()`
Addressed here: https://github.com/alpacahq/quickfix/pull/17/commits/c835cd472ea2f2bf51094bfab76420905d79d0be

Reply:
Try changing the `CleanIncomingHotPath` variable true/false

Reply:
CleanIncomingHotPath = true
```
/private/var/folders/cp/r9_v98l942x0jxt4d4fqyry80000gn/T/GoLand/___2BenchmarkIncoming_in_github_com_quickfixgo_quickfix.test -test.v -test.paniconexit0 -test.bench ^\QBenchmarkIncoming\E$ -test.run ^$
goos: darwin
goarch: arm64
pkg: github.com/quickfixgo/quickfix
BenchmarkIncoming
BenchmarkIncoming-10    	   83115	     13667 ns/op	    7983 B/op	      78 allocs/op
PASS

```

CleanIncomingHotPath = false
```
/private/var/folders/cp/r9_v98l942x0jxt4d4fqyry80000gn/T/GoLand/___2BenchmarkIncoming_in_github_com_quickfixgo_quickfix.test -test.v -test.paniconexit0 -test.bench ^\QBenchmarkIncoming\E$ -test.run ^$
goos: darwin
goarch: arm64
pkg: github.com/quickfixgo/quickfix
BenchmarkIncoming
BenchmarkIncoming-10    	   14322	     84926 ns/op	   40275 B/op	     455 allocs/op
PASS
```